### PR TITLE
Add 39 failing bitwise test scripts

### PR DESCRIPTION
**Pull request by Unisay**
## Summary

Implements generation of 39 failing bitwise test scripts to restore complete test coverage for Plutus Core 1.1.0 bitwise primitives. This addresses the gap created during the November 2025 library/executable separation refactoring.

**Impact**: Increases generated scripts from 20 to 59 (restores 1:1 correspondence with cardano-node-tests for bitwise tests)

### ScriptGroup Infrastructure

Introduced a generic `ScriptGroup` type in `Helpers/ScriptUtils.hs`:
- Organizes parameterized test variants with clean separation of concerns
- Library handles parameter iteration, executable generates numbered files
- Reusable pattern for any multi-variant test scenarios

### Failing Test Coverage (39 new scripts)

**ReadBit Tests** (`failingReadBitPolicyScriptV3_1` through `_14`):
- Empty bytestring edge cases
- Negative index validation
- Out of bounds access
- Int64 min/max boundary conditions

**WriteBits Tests** (`failingWriteBitsPolicyScriptV3_1` through `_19`):
- Empty bytestring edge cases
- Negative indices in bit position lists
- Out of bounds bit positions
- Large indices beyond Int64 range

**ReplicateByte Tests** (`failingReplicateBytePolicyScriptV3_1` through `_6`):
- Negative count validation
- Invalid byte values (< 0, > 255)
- Size limit violations (exceeding 8192 byte maximum)

### Architecture Maintained

✅ **Library stays pure**: Only exports `CompiledCode` values, no IO operations
✅ **Params encapsulation**: Iteration via `map` in library, executable unaware of Params types
✅ **Clean separation**: Library compiles, executable writes files
✅ **Type safety**: ScriptGroup ensures base name and scripts stay together

### Files Modified

- `Helpers/ScriptUtils.hs`: Added ScriptGroup type definition
- `PlutusScripts/Bitwise/V_1_1.hs`: Created failingBitwiseScriptGroupsV3 export
- `app/Main.hs`: Added writeScriptGroup helper and generation call
- `plutus-scripts.cabal`: Added plutus-core dependency to executable
- `README.md`: Updated script counts and failing test descriptions
- `CLAUDE.md`: Documented ScriptGroup pattern with examples

## Testing

Build verification:
```bash
cabal build plutus-scripts --ghc-options=-Werror  # ✅ Passes
cabal run envelopes  # ✅ Generates 59 scripts
```

Output verification:
- Total scripts: 59 (previously 20)
- Failing ReadBit: 14 scripts
- Failing WriteBits: 19 scripts
- Failing ReplicateByte: 6 scripts
- All naming matches cardano-node-tests format

## Verification Against cardano-node-tests

**File Presence**: ✅ All 39 failing test scripts present with correct naming
**CBOR Comparison**: Different encodings (expected - compiled with Plutus 1.53 vs 1.36 reference)
**File Sizes**: Comparable, often smaller (suggests compiler improvements)

### CBOR Differences

All scripts have different CBOR encodings compared to cardano-node-tests reference scripts. This is **expected and acceptable** because:
- Reference scripts compiled with Plutus 1.36 (October 2024)
- Our scripts compiled with Plutus 1.53 (November 2025)
- Different compiler versions produce functionally equivalent but differently encoded bytecode
- Our scripts are generally smaller (better optimization)

Functional equivalence should be validated by running cardano-node-tests with our generated scripts.

### Out of Scope

The following scripts exist in cardano-node-tests but are not added in this PR:
- `blsMint.plutus`
- `blsSpend.plutus`
- `constitutionScriptV3.plutus`

Git history shows these were **never generated by this repository** (BLS and Governance modules exist as library code only). Awaiting clarification on whether to add them.

## Related

Closes investigation into missing failing test scripts documented in commit history.

### CODE REVIEW

File: plutus-scripts/PlutusScripts/Bitwise/V_1_1.hs [hunk 1]
```diff
@@ -30,10 +30,12 @@ import PlutusScripts.Bitwise.Logical (
   succeedingXorByteStringParams,
  )
 import PlutusScripts.Bitwise.ReadBit (
+  failingReadBitParams,
```
Comment (kwxm) [inline plutus-scripts/PlutusScripts/Bitwise/V_1_1.hs:33]: These big import lists take up a lot of room and maybe it'd be more readable if we qualified them.  It's not that big a deal though.

File: plutus-scripts/PlutusScripts/Bitwise/V_1_1.hs [hunk 2]
```diff
@@ -30,10 +30,12 @@ import PlutusScripts.Bitwise.Logical (
...
   mkReadBitPolicy,
   succeedingReadBitParams,
  )
 import PlutusScripts.Bitwise.ReplicateByte (
+  failingReplicateByteParams,
   mkReplicateBytePolicy,
   succeedingReplicateByteParams,
```
Comment (kwxm) [inline plutus-scripts/PlutusScripts/Bitwise/V_1_1.hs:40]: I didn't know that we were allowed trailing commas at the end of import lists, and I don't think we use them anywhere else.  Is that something that's configurable in Fourmolu?

File: plutus-scripts/PlutusScripts/Bitwise/V_1_1.hs [hunk 3]
```diff
@@ -149,6 +152,38 @@ succeedingReplicateBytePolicyCompiledV3 =
...
+--------------------------------------------------------------------------------
+-- Failing Tests ---------------------------------------------------------------
+
+-- | All failing bitwise test script groups
+-- Each group generates numbered scripts (e.g., failingReadBit_1.plutus, failingReadBit_2.plutus)
+failingBitwiseScriptGroupsV3 :: [ScriptGroup DefaultUni DefaultFun (P.BuiltinData -> P.BuiltinUnit)]
+failingBitwiseScriptGroupsV3 =
+  [ -- ReadBit failing tests
+    ScriptGroup
+      { sgBaseName = "failingReadBitPolicyScriptV3"
```
Comment (kwxm) [inline plutus-scripts/PlutusScripts/Bitwise/V_1_1.hs:164]: Do we really need the field names here?

**Review by kwxm — APPROVED** [review]
I had a few minor quibbles about formatting and style, but I don't think it's worth holding up the merge while we have a long debate!  Let's merge this and move on.

File: plutus-scripts/app/Main.hs [hunk 4]
```diff
@@ -5,13 +5,29 @@ import Main.Utf8 (withUtf8)
 import PlutusLedgerApi.Common.Versions (PlutusLedgerLanguage (PlutusV2, PlutusV3))
 import PlutusLedgerApi.Envelope qualified as Envelope
 import PlutusScripts.Basic.V_1_1 qualified as Basic
+import Control.Monad (zipWithM_)
+import Helpers.ScriptUtils (ScriptGroup (ScriptGroup, sgBaseName, sgScripts))
+import PlutusCore.Default (DefaultFun, DefaultUni)
```
Comment (kwxm) [inline plutus-scripts/app/Main.hs:10]: I don't remember if it was like this before the refactoring, but things like `import PlutusScripts.Bitwise.V_1_1 qualified as BitwiseV1` are a bit confusing because `V1` usually refers to PlutusV1.  I'm not sure what would be better though.

File: plutus-scripts/PlutusScripts/Bitwise/V_1_1.hs (hunk 2 above)
Comment (Unisay) [reply plutus-scripts/PlutusScripts/Bitwise/V_1_1.hs:40]: Yes
